pkg/aggregator/check_sampler.go

```diff
@@ -7,44 +7,38 @@ package aggregator
 
 import (
 	"math"
-	"time"
 
 	"github.com/DataDog/datadog-agent/pkg/aggregator/ckey"
 	"github.com/DataDog/datadog-agent/pkg/metrics"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-const defaultExpiry = 300.0 // number of seconds after which contexts are expired
 const checksSourceTypeName = "System"
 
 // CheckSampler aggregates metrics from one Check instance
 type CheckSampler struct {
 	series          []*metrics.Serie
 	sketches        metrics.SketchSeriesList
-	contextResolver *ContextResolver
+	contextResolver *countBasedContextResolver
 	metrics         metrics.ContextMetrics
```

Comment:
I think we also want to expire the entries in `metrics.ContextMetrics` for the contexts that are expired. Until we do that, I think checks that submit an unbounded number of contexts across check runs will still cause a memory leak. We can work on fixing this as a further improvement for 7.30. Let's just update the PR description for now.

Note that when we start expiring entries in this map, we'll actually likely want to set a higher default than `1` for the expiration. I just saw that https://github.com/DataDog/datadog-agent/pull/8268 would make the `docker` check sometimes not submit a value for some rate metrics during some check runs. Increasing the value to `2` would allow the aggregator to not expire the rates too early (in scenarios where #8268 only makes the docker check not submit rates for at most one check run in a row).

Reply:
> I think we also want to expire the entries in metrics.ContextMetrics for the contexts that are expired. Until we do that, I think checks that submit an unbounded number of contexts across check runs will still cause a memory leak. We can work on fixing this as a further improvement for 7.30. Let's just update the PR description for now.

Tracked by AC-1011

> Note that when we start expiring entries in this map, we'll actually likely want to set a higher default than 1 for the expiration. I just saw that #8268 would make the docker check sometimes not submit a value for some rate metrics during some check runs. Increasing the value to 2 would allow the aggregator to not expire the rates too early (in scenarios where #8268 only makes the docker check not submit rates for at most one check run in a row).

Fixed by https://github.com/DataDog/datadog-agent/pull/8392